File: PSC/Series/Serie-04/Documents/java/NoRepValidator.java

public final class NoRepValidator implements ExamValidator.TwoExamsValidator {
	public String ruleName() { return "Sobreposicoes nao se repetem nas duas epocas."; }
	public void setArgs(String dist) {  }
	public boolean isValid(Exam exam1, Exam exam2) {
		return (exam1.date1() != exam2.date1()) ||
		       (exam1.date2() != exam2.date2());
	}
}

File: PSC/Series/Serie-04/src/NoRepValidator.c
#include <stdio.h>
#include "myLib.h"
#include "NoRepValidator.h"


static nrvMethods nrv_vtable={
	dtor,
	ruleName,
	setArgs,
	isValid
};

void* ctor(){
	nrv* n = (nrv*)malloc(sizeof(nrv));
	n->vptr = &nrv_vtable;                     
	return (void*)(n);
}

void dtor(twoEV* this){
	if (this != NULL)
		free(this);
}

void setArgs (twoEV* this, String dest){
}

String ruleName (){
	return "Sobreposicoes nao se repetem nas duas epocas.";
}

int isValid(twoEV* this, exam *exam1, exam *exam2){
	return (getDate1(exam1) != getDate1(exam2) ||
			getDate2(exam1) != getDate2(exam2));
}





